lib/named-locks.js

```diff
@@ -1,117 +1,153 @@
+// @ts-check
 const ERR = require('async-stacktrace');
 const async = require('async');
+const util = require('util');
 
 const sqldb = require('@prairielearn/prairielib/sql-db');
 const sqlLoader = require('@prairielearn/prairielib/sql-loader');
 
 const sql = sqlLoader.loadSqlEquiv(__filename);
 
-/*
- * The lock object returned by functions in this module are of the
- * form `{client, done}`, where `client` and `done` are sqldb
- * transaction objects.
+/** @typedef {{ client: import("pg").PoolClient, done: (release?: any) => void }} Lock */
+/** @typedef {{ timeout?: number }} Options */
+
+/**
+ * Try to acquire a lock and either succeed if it's available or
+ * return immediately if not. If a lock is acquired then it must
+ * later be released by releaseLock().
+ *
+ * @param {string} name The name of the lock to acquire.
+ * @param {(err: Error | null, lock: Lock) => void} callback A callback(err, lock) function. If lock is null then it was not acquired.
  */
+module.exports.tryLock = function(name, callback) {
+    _getLock(name, { wait: false }, (err, lock) => {
+        if (ERR(err, callback)) return;
+        callback(null, lock);
+    });
+};
 
-module.exports = {
-    /**
-     * Try to acquire a lock and either succeed if it's available or
-     * return immediately if not. If a lock is acquired then it must
-     * later be released by releaseLock().
-     *
-     * @param {string} name - The name of the lock to acquire.
-     * @param {function} callback - A callback(err, lock) function. If lock is null then it was not acquired.
-     */
-    tryLock(name, callback) {
-        this._getLock(name, {wait: false}, (err, lock) => {
-            if (ERR(err, callback)) return;
-            callback(null, lock);
-        });
-    },
+module.exports.tryLockAsync = util.promisify(module.exports.tryLock);
```

Comment:
Following up on the resolved comment above, there are more of these `...Async()` versions without their own docs. I’m not necessarily pushing to add docs here, just pointing it out in case you missed them 😄